nix setup for running test.sh locally

pmbittner opened the pull request:
With nix, you do not have to install python or the required python packages if you have nix installed. Just open a nix-shell and run the test.sh script.

The commits of this PR should probably be **squashed** when merged int `main` because they contain some experimentation.

skrieter reviewed (approved):
Looks fine to me, but I am no expert at nix.

pmbittner commented:
Maybe @ibbem should have a look. :)

pmbittner commented:
Hi @ibbem ,
thanks for your review and the excellent comments.
1. `with`-syntax: Good point, I removed the `with`-clause.
2. shebang: We already fixed this in some other scripts but I guess I forgot this location.
3. purity: Oops. Thanks for pointing that out. Apparently, the jdk is not the only missing dependency, we also must include LaTeX :D. I added `texliveFull` as well. (It seems there is no smaller nix package for LaTeX that satisfies all dependencies, I tried `texliveSmall` and `texliveMedium`). Maybe we could try to specify only the packages we actually need to keep the environment as small as possible.

ibbem commented:
I just remembered that we can just use `nix-shell --pure` to check for missing dependencies. 🤦
This way I confirmed that this was indeed the last missing dependency. 😄

pmbittner commented:
Ok thanks. :) That is what I did too.
